node/actors/bft/src/leader/replica_commit.rs

```diff
@@ -127,11 +127,22 @@ impl StateMachine {
             .or_insert_with(|| CommitQC::new(message.clone(), self.config.genesis()))
             .add(&signed_message, self.config.genesis());
 
-        // We store the message in our cache.
         let cache_entry = self
             .commit_message_cache
             .entry(message.view.number)
             .or_default();
+
+        // We check validators weight from current messages, stored by proposal
+        let mut by_proposal_before: HashMap<_, Vec<_>> = HashMap::new();
+        let entry_before = cache_entry.clone();
+        for msg in entry_before.values() {
+            by_proposal_before
+                .entry(msg.msg.proposal)
```

Comment:
you can just compute the weight of the proposal of the new message, rather than for all proposals.

Reply:
Ah, right!

Reply:
And actually we can do the same in later weight calculation